packages/paste-core/components/button/src/index.tsx

```diff
@@ -69,34 +89,40 @@ const Button: React.FC<ButtonProps> = props => {
     }
   }
 
-  handlePropValidation(props);
+  handlePropValidation({as, href, variant, size, tabIndex, fullWidth, children});
 
   return (
     <ButtonWrapper
-      as={props.as}
-      variant={props.variant}
-      size={defaultSize}
-      buttonState={buttonState}
-      disabled={disabled}
-      autoFocus={props.autoFocus}
-      fullWidth={props.fullWidth}
-      href={props.href}
-      tabIndex={props.tabIndex}
-      type={props.type}
-      onClick={props.onClick}
-      onMouseDown={props.onMouseDown}
-      onMouseUp={props.onMouseUp}
-      onMouseEnter={props.onMouseEnter}
-      onMouseLeave={props.onMouseLeave}
-      onFocus={props.onFocus}
-      onBlur={props.onBlur}
+      aria-busy={buttonState === 'loading' ? 'true' : 'false'}
+      aria-controls={props['aria-controls']}
       aria-expanded={props['aria-expanded']}
       aria-haspopup={props['aria-haspopup']}
-      aria-controls={props['aria-controls']}
-      aria-busy={buttonState === 'loading' ? 'true' : 'false'}
-      data-test={props['data-test']}
+      as={as}
```

Comment:
For any of these that we actually don’t use in the component scope or logic and just passing down, lets just rely on the spreading of props instead of destructuring them off.

It’ll make this look a little tidier